libHVCAENx527App/src/HVCAENx527.c

```diff
@@ -61,6 +62,45 @@ epicsThreadId scanThread;
 
 HVCRATE Crate[MAX_CRATES];
 
+static epicsMutexId busyMutexId; /* to protect access to Busy[] */
+static int busyInitDone = 0;
+
+static void busyLock(int crate)
+{
+    int done = 0;
+    if (crate < 0 || crate >= MAX_CRATES)
+    {
+        printf("busyLock: crate %d out of range \n", crate);
```

Comment:
Should this be `epicsErrlogPrintf` so that it goes to the error log and database etc?

Reply:
Possibly - i was just following the convention in the rest of the file which seems to use printf() for such things, not to say that is right of course. I'm not sure if the check is really needed, i was just checking for every possible thing when we were debugging a once in two weeks hang...